### DISCUSSION TITLE
RP2-PICO: USB MSC volume label

### DISCUSSION THREAD

**Discussion by josch0** [micropython/micropython#9590] [Core Development]
Hello all,

I am using my own firmware (RP2 PICO) with MSC support. The actual python application is frozen in the firmware, I only use MSC to make a config file easily editable for the user.

To allow the user to find the USB drive quickly, I would like to give the volume a predefined label. So far I haven't figured out how to do that.

Can someone give me a hint?

**Comment by robert-hh** [comment]
One it is mounted to a PC, you can give the drive a name, using the PC tools.

**Reply by josch0** [reply]
Hi Robert!

Yes I know this way.

But the name should be set automatically by the firmware. Just like the PICO appears automatically as "RPI-RP2" in BOOTSEL mode.

**Comment by robert-hh** [comment]
I do not see that it can be changed by a MicroPython API. The name "NO NAME" is hard coded in lib/oofatfs/ff.c, lines 5739 or 5735. Since you build the code yourself, you can change it to a different fixed start name.

**Reply by jimmo** [reply]
@josch0 In more detail, what's going on here is that the default filesystem is created by `_boot_fat.py` which uses the Python API to create the filesystem. Like Robert says, there's no Python API to set the label.

On STM32, the default filesystem is created in C, where it can use `f_setlabel()` directly (and it does this using `MICROPY_HW_FLASH_FS_LABEL` which optionally is set by the board in mpconfigboard.h).

If you want to make this work, you could add a `label()` method to `extmod/vfs_fat.c` and then call that from _boot.py. (Or just hard code a call to f_setlabel) anywhere in vfs_fat.c that does a `f_mkfs`.
